GitUI/FilterBranchHelper.cs

```diff
@@ -143,18 +144,35 @@ private void toolStripBranches_DropDown(object sender, EventArgs e)
 
         private void ApplyBranchFilter(bool refresh)
         {
-            bool success = _NO_TRANSLATE_RevisionGrid.SetAndApplyBranchFilter(_NO_TRANSLATE_toolStripBranches.Text);
-            if (success && refresh)
-                _NO_TRANSLATE_RevisionGrid.ForceRefreshRevisions();
+            if (_applyingFilter)
```

Comment:
Not fully threadsafe, but should be good enough

Reply:
I thought of it, but the helper is used from FormBrowse on the UI thread, so there can't be any x-treading.